app/forms/degree_form.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+class DegreeForm
+  include ActiveModel::Model
+  include ActiveModel::AttributeAssignment
+  include ActiveModel::Validations::Callbacks
+
+  FIELDS = %i[
+    slug
+    locale_code
+    uk_degree
+    non_uk_degree
+    subject
+    institution
+    graduation_year
+    grade
+    other_grade
+    country
+  ].freeze
+
+  attr_accessor(*FIELDS, :parent, :degree)
+
+  validate :validate_with_degree_model
+
+  delegate :uk?, :non_uk?, :non_uk_degree_non_naric?,
+    to: :degree
+
+  def initialize(parent:, degree:)
+    @parent = parent
+    @degree = degree
+    self.attributes = degree.attributes
+      .symbolize_keys
+      .slice(*FIELDS)
+  end
+
+  def persisted?
+    slug.present?
+  end
+
+  def id
+    slug
+  end
+
+  def fields
+    degree.attributes
+      .symbolize_keys
+      .slice(*FIELDS)
+      .merge(attributes)
+  end
+
+  def attributes
+    FIELDS.inject({}) do |sum, f|
+      sum[f] = send(f)
+      sum
+    end
+  end
+
+  def attributes=(attrs)
+    attrs.each do |k,v|
+      send(:"#{k}=", v)
+    end
+  end
+
+  def save_or_stash
+    if parent.trainee.draft?
+      save!
+    else
+      stash
+    end
+  end
+
+  def stash
+    return false unless valid?
```

Comment:
I think `valid? && parent.stash_degree_on_store(slug, fields)` gives the same outcome